### PR TITLE
Added network integration test, fixed autoroute binary generator so t…

### PR DESCRIPTION
**Pull request by dcrck**
…hat 'listen' is no longer required. Also deleted the last of the shell scripts.

### CODE REVIEW

**Comment by c00w** [comment]
Pass some packets through and make sure they make it?

**Comment by dcrck** [comment]
Do we already have functions to do this?

**Comment by dcrck** [comment]
If not, we should have it.

**Comment by c00w** [comment]
Yeah see the packet test.

**Comment by dcrck** [comment]
Ahh, gotcha, will fix when Eric merges your PR.

**Comment by dcrck** [comment]
Do you need to establish the sockets before routing packets? I don't have "/tmp/unix" or "/tmp/unix2" on my system. What do they do on your system?

**Comment by c00w** [comment]
Those sockets are established by the autoroute binary. They're just a hedge
to make it easy to route packets into the system.

**Comment by dcrck** [comment]
So I added most of the code from the packet test, but one WaitForPacketsReceived (see my comments in the code for this last commit) fails for some reason. The other ones work just fine. Do you see any reasons behind this?

**Comment by c00w** [comment]
Off the top of my head no. I'll look tomorrow?

**Comment by dcrck** [comment]
For packets sent and received: how does "amount" change over time? Each packet has an id and an "amount." The first time I wait for a packet at that specific "node" the value of all packets received is 1. The second time I read from that ID, the value of all packets is 2. I think that may be the issue: I'm always waiting for a value of 1, but the value actually changes to 2.

**Comment by c00w** [comment]
In the exported map we just count how many packets we've seen. So if you
send 2 packets, it'll say 2.

**Comment by c00w** [comment]
It mostly for debugging since you can listen at the end for the specific
packet.

**Comment by dcrck** [comment]
Do I need more than one unix socket for a binary if the binary has multiple interfaces/devs?

**Comment by dcrck** [comment]
Running the test sometimes works and sometimes does not, and the reason for failure is different each time. I guess that means there are larger problems here.

**Comment by c00w** [comment]
Did you remember to delete the socket?

**Comment by dcrck** [comment]
So I added some print statements, and I'm getting a connection error. Both binaries show the connection receving though.

For example: http://pastebin.com/rV89gCzd

**Comment by c00w** [comment]
Ah, so this is likely a ordering / race condition. If you configure one device to connect to another, it'll only attempt it once, if the other binary isn't up yet it might fail. Since your nodes ultimately connect, I'm guessing that the timeout bounds are too low, or you have problems with not bringing one of the binaries up until after you check for a connection. Since you're doing a network simulation, I'd just set auto discover for each node and not worry about manually configuring connections. It'll figure it out. If you really care, I'd verify the connections after you've started all of the binaries.

**Comment by c00w** [comment]
And to answer your comment, no you shouldn't need more than 1 unix socket per node. It just provides a convenient injection interface.

**Comment by dcrck** [comment]
Autodiscover is on for each node...and I start all the binaries before attempting to verifying connections
Added more print statements. It appears that the "connections" are only the "outgoing connections" and not the "established connections" ( I printed out the connections during WaitForConnections any time the connections list is not empty). For example: http://pastebin.com/yTcD8EDJ

Any idea why this could be?

**Comment by dcrck** [comment]
It might have to do with the FetchStatus function?

File: autoroute/main.go [hunk 1]
```diff
@@ -53,13 +53,13 @@ func Probe(key node.PrivateKey, n *node.Server, dev net.Interface, port uint16)
 
 	neighbors := node.FindNeighbors(dev, ll_addr, key.PublicKey(), port)
 	for neighbor := range neighbors {
-		log.Printf("Neighbour Found %x", neighbor.NodeAddr)
+		log.Printf("Neighbour Found %x", neighbor.NodeAddr[0:4])
```
Comment (c00w) [inline autoroute/main.go]: Remove this?

File: integration_tests/test_helpers.go [hunk 2]
```diff
@@ -73,8 +73,14 @@ func WaitForPacketsReceived(b AutoRouteBinary, src string, amt int) error {
...
 		}
 		for source, amount := range packets_received {
-			if source == src && amount == amt {
-				return nil
+			if len(amt) > 0{
+				if source == src && amount == amt[0] {
+					return nil
+				}
+			} else {
+				if source == src {
```
Comment (c00w) [inline integration_tests/test_helpers.go:81]: Uh I'm 90% sure that this isn't checking any packets have been received.

File: integration_tests/test_helpers.go [hunk 3]
```diff
@@ -87,7 +93,7 @@ func WaitForPacketsReceived(b AutoRouteBinary, src string, amt int) error {
 	panic("unreachable")
 }
 
-func WaitForPacketsSent(b AutoRouteBinary, dest string, amt int) error {
+func WaitForPacketsSent(b AutoRouteBinary, dest string, amt ...int) error {
```
Comment (c00w) [inline integration_tests/test_helpers.go:96]: Doesn't actually check that packets are sent

File: integration_tests/test_helpers.go (hunk 2 above)
Comment (c00w) [inline integration_tests/test_helpers.go:81]: NVM I'm insane, we should probably check amount > 0 though.

File: integration_tests/test_helpers.go (hunk 3 above)
Comment (c00w) [inline integration_tests/test_helpers.go:96]: NVM I'm insane.

File: integration_tests/test_helpers.go [hunk 4]
```diff
@@ -95,8 +101,14 @@ func WaitForPacketsSent(b AutoRouteBinary, dest string, amt int) error {
...
 		}
 		for destination, amount := range packets_sent {
-			if destination == dest && amount == amt {
-				return nil
+			if len(amt) > 0{
+				if destination == dest && amount == amt[0] {
+					return nil
+				}
+			} else {
+				if destination == dest {
```
Comment (c00w) [inline integration_tests/test_helpers.go:114]: Check amount > 0?

**Comment by dcrck** [comment]
Done

**Pull request merged**